src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Asynchrony.cs

```diff
@@ -26,6 +26,11 @@ public abstract class HttpClientHandler_Asynchrony_Test : HttpClientHandlerTestB
         [MemberData(nameof(ResponseHeadersRead_SynchronizationContextNotUsedByHandler_MemberData))]
         public async Task ResponseHeadersRead_SynchronizationContextNotUsedByHandler(bool responseHeadersRead, LoopbackServer.ContentMode contentMode)
         {
+            if (IsWinHttpHandler && (PlatformDetection.IsWindows7 || PlatformDetection.IsWindows8x))
+            {   // [ActiveIssue("https://github.com/dotnet/runtime/issues/54034")]
+                throw new SkipTestException("Win7/Win81 environment often hangs the test for WinHttpHandler.");
```

Comment:
I think you will have to change `[Theory]` to `[ConditionalTheory]` to get the `SkipTestException` working properly.

Reply:
I used return instead, that works too ...